source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4196,6 +4196,40 @@ export class Choice extends SerializableObject {
     }
 }
 
+/**
+ * DataQuery class is declared later in the file and derives from subsequent base classes
+ * Hence, it cannot be used in ChoiceSetInput.
+ * Refactor is needed to separate elements and actions in separate files.
+ */
+export class ChoiceSetInputDataQuery extends SerializableObject {
+    //#region Schema
+
+    static readonly typeProperty = new StringProperty(
+        Versions.v1_6,
+        "type",
+        true,
+        new RegExp("/^Data.Query$/")
+    );
+    static readonly datasetProperty = new StringProperty(Versions.v1_6, "dataset");
+
+    @property(ChoiceSetInputDataQuery.typeProperty)
+    type: string;
+
+    @property(ChoiceSetInputDataQuery.datasetProperty)
+    dataset: string;
+
+    //#endregion
+
+    protected getSchemaKey(): string {
+        return "choices.data";
+    }
+}
+
+export type FetchedChoice = {
```

Comment:
@jwoo-msft Could we use `Choice` here instead? Or do we want to avoid that since it extends `SerializableObject`?

Reply:
Precisely. We can do it using `Choice` as well. But this will require additional processing upon receiving the raw response from the host, since `Choice` is not just about `title` and `string` but has the properties of a `SerializableObject`, and we do not need that here.